tools/clang/lib/SPIRV/SpirvInstruction.cpp

```diff
@@ -987,5 +988,12 @@ SpirvReadClock::SpirvReadClock(QualType resultType, SpirvInstruction *s,
     : SpirvInstruction(IK_ReadClock, spv::Op::OpReadClockKHR, resultType, loc),
       scope(s) {}
 
+SpirvRayTracingTerminateOpKHR::SpirvRayTracingTerminateOpKHR(spv::Op opcode,
+                                                             SourceLocation loc)
+    : SpirvTerminator(IK_RayTracingTerminate, opcode, loc) {
+  assert(opcode == spv::Op::OpTerminateRayKHR ||
+         spv::Op::OpIgnoreIntersectionKHR);
```

Comment:
```suggestion
         opcode == spv::Op::OpIgnoreIntersectionKHR);
```

Praise clang for catching this one:
```c
../tools/clang/lib/SPIRV/SpirvInstruction.cpp:994:47: warning: converting the enum constant to a boolean [-Wint-in-bool-context]
  assert(opcode == spv::Op::OpTerminateRayKHR ||
                                              ^
../tools/clang/lib/SPIRV/SpirvInstruction.cpp:995:10: error: value of type 'spv::Op' is not contextually convertible to 'bool'
         spv::Op::OpIgnoreIntersectionKHR);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
ouch. Sad that MSVC didn't catch this. Also not sure why Appveyor bot is passing.

Reply:
I know why appveyor is passing... MSVC doesn't catch the build warning and the FileCheck tests run without optimizations.

Reply:
Ok, we need to update SPIRV-Tools to recognize these instructions as terminator as well :) Will do so soon

Reply:
Another explanation would be Clang version. There are a few new warnings on Clang 11 but Appveyor is still at 10.0.0.1.

Reply:
We should land this PR, and we will separately make fixes to SPIRV-Tools for recognizing these instructions as termination instructions.